src/extension/geofence/geofence/src/main/java/org/geoserver/geofence/RuleFilterBuilder.java

```diff
@@ -0,0 +1,225 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.geofence;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.commons.lang.StringUtils;
+import org.geoserver.geofence.config.GeoFenceConfiguration;
+import org.geoserver.geofence.services.dto.RuleFilter;
+import org.geoserver.ows.Request;
+import org.geotools.util.logging.Logging;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+
+/** Builder class for a {@link RuleFilter}. */
+class RuleFilterBuilder {
```

Comment:
This class is unit testable, I'd add a few checks on it.

Reply:
test added